cigaradvisor/scripts/scripts.js

```diff
@@ -203,57 +209,66 @@ export function getRelativePath(path) {
  */
-export async function fetchData(filterPath, fetchUrl = '/query-index.json') {
-  let responeData = '';
-  if (fetchUrl === '/query-index.json') {
-    if (!indexData) {
-      const resp = await fetch(fetchUrl);
-      let jsonData = '';
-      if (resp.ok) {
-        jsonData = await resp.json();
-      }
-      indexData = jsonData.data;
+export async function fetchPostsInfo(filterValue, filterParam = 'path') {
+  let filter = filterValue;
+  filter = getRelativePath(filterValue);
+  if (!articleIndexData) {
+    const resp = await fetch(ARTICLE_INDEX_PATH);
+    let jsonData = '';
+    if (resp.ok) {
+      jsonData = await resp.json();
     }
-    responeData = indexData;
+    articleIndexData = jsonData.data;
   }
-  if (fetchUrl === '/cigaradvisor/posts/query-index.json') {
-    if (!articleIndexData) {
-      const resp = await fetch(fetchUrl);
-      let jsonData = '';
-      if (resp.ok) {
-        jsonData = await resp.json();
-      }
-      articleIndexData = jsonData.data;
+  const filteredData = articleIndexData.filter((obj) => obj[filterParam] === filter);
+  return filteredData;
+}
+
+let authorIndexData = '';
+/**
+ * Fetches author information based on the provided author link.
+ * @param {string} authorLink - The link to the author's page.
+ * @returns {Promise<Array<Object>>} - A promise that resolves to an array of
+ * author information objects.
+ */
+export async function fetchAuthorInfo(authorLink) {
+  const filter = getRelativePath(authorLink);
+  if (!authorIndexData) {
+    const resp = await fetch(AUTHOR_INDEX_PATH);
+    let jsonData = '';
+    if (resp.ok) {
+      jsonData = await resp.json();
     }
-    responeData = articleIndexData;
+    authorIndexData = jsonData.data;
   }
-  if (fetchUrl === '/cigaradvisor/author/query-index.json') {
-    if (!authorIndexData) {
-      const resp = await fetch(fetchUrl);
-      let jsonData = '';
-      if (resp.ok) {
-        jsonData = await resp.json();
-      }
-      authorIndexData = jsonData.data;
+  return authorIndexData.filter((obj) => obj.path === filter);
+}
+
+let categoryIndexData = '';
+/**
+ * Fetches category information based on the provided category link.
+ * @param {string} categoryLink - The link to the category.
+ * @returns {Promise<Array<Object>>} - A promise that resolves to an array of
+ * category objects matching the provided link.
+ */
+export async function fetchCategoryInfo(categoryLink) {
+  const filter = getRelativePath(categoryLink);
+  if (!categoryIndexData) {
+    const resp = await fetch(DEFAULT_INDEX_PATH);
+    let jsonData = '';
+    if (resp.ok) {
+      jsonData = await resp.json();
     }
-    responeData = authorIndexData;
+    categoryIndexData = jsonData.data;
   }
-  let filteredData = '';
-  if (responeData) {
-    filteredData = responeData.find((obj) => obj.path === filterPath);
-  }
-  return filteredData;
+  return categoryIndexData.filter((obj) => obj.path === filter);
```

Comment:
rather than return the filtered array, return the single entry,so that caller's don't have to check the arrary contents

it's either the object, or undefined

Reply:
Done